al_aws_collector.js

```diff
@@ -378,6 +402,28 @@ class AlAwsCollector {
             }
         ],
         function(err, checkinParts) {
+
+            let invocationStatsDatapoints = checkinParts[1].statistics[0].Datapoints ? checkinParts[1].statistics[0].Datapoints : checkinParts[1].statistics;
+            let errorStatsDatapoints = checkinParts[1].statistics[1].Datapoints ? checkinParts[1].statistics[1].Datapoints : checkinParts[1].statistics ;
+
+            if (checkinParts[0].status === 'ok'  && invocationStatsDatapoints.length > 0  && invocationStatsDatapoints[0].Sum > 0
+                 && errorStatsDatapoints.length > 0 && errorStatsDatapoints[0].Sum === 0) {
+                let collectorStreams = JSON.parse(collector._streams);
```

Comment:
Is `collector._streams` a JSON object? It is passed as list into the constructor. 
If we really need to parse streams from JSON object let's do this once in the constructor.

Reply:
Yes we have to parse the collector_streams as we are using env variable with value [\"AuditLogEvents\", \"SearchAlerts\"]". I will  update and parse this in constructor.